src/assets/wise5/components/component-student.component.ts

```diff
@@ -9,18 +17,463 @@ export abstract class ComponentStudent {
   @Input()
   componentContent: any;
 
+  @Input()
+  componentState: any;
+
+  @Input()
+  mode: string;
+
+  @Input()
+  workgroupId: number;
+
   componentId: string;
+  componentType: string;
+  prompt: string;
+  isPromptVisible: boolean = true;
+  isSaveButtonVisible: boolean = false;
+  isSubmitButtonVisible: boolean = false;
+  isSaveOrSubmitButtonVisible: boolean = false;
+  isSubmitButtonDisabled: boolean = false;
+  isSubmitDirty: boolean = false;
+  isSubmit: boolean = false;
+  isDirty: boolean = false;
+  isDisabled: boolean = false;
+  submitCounter: number = 0;
+  latestAnnotations: any;
+  saveMessage = {
+    text: '',
+    time: ''
+  };
+  requestComponentStateSubscription: Subscription;
+  annotationSavedToServerSubscription: Subscription;
+  nodeSubmitClickedSubscription: Subscription;
+  studentWorkSavedToServerSubscription: Subscription;
```

Comment:
```suggestion
  subs: Subscription;
```
Use Subscription.add and Subscription.unsubscribe to simplify code. Also change code in registerListeners() to add to this subs object.